packages/@aws-cdk/aws-codepipeline-actions/lib/stepfunctions/invoke-action.ts

```diff
@@ -0,0 +1,155 @@
+import * as codepipeline from '@aws-cdk/aws-codepipeline';
+import * as iam from '@aws-cdk/aws-iam';
+import * as stepfunction from '@aws-cdk/aws-stepfunctions';
+import { Construct } from '@aws-cdk/core';
+import { Action } from '../action';
+
+/**
+ * Represents the input for the StateMachine.
+ */
+export class StateMachineInput {
+  /**
+   * When the input type is FilePath, input artifact and
+   * filepath must be specified.
+   */
+  public static filePath(filePath: string, inputArtifact: codepipeline.Artifact): StateMachineInput {
```

Comment:
Apologies @madhumitranrl , but I came up with an even better API :). We have a class in the CodePipeline library exactly for this (file in an artifact) purpose: [`ArtifactPath`](https://github.com/aws/aws-cdk/blob/92e620c5ee4262736d5fa576193793d7771ec47c/packages/%40aws-cdk/aws-codepipeline/lib/artifact.ts#L123-L123). So, you can change this function to:

```ts
  public static filePath(inuputFile: codepipeline.ArtifactPath): StateMachineInput {
    return new StateMachineInput(inuputFile.location, inuputFile.artifact, 'FilePath');
  }
```

And the customer uses it like this:

```ts
StateMachineInput .filePath(buildArtifact.atPath('path/to/my/file.json'));
```

This will be perfect!

Now, with this design, what do you think about getting rid of `StateMachineInput` class? We can change `StepFunctionsInvokeActionProps` to be just:

```ts
export interface StepFunctionInvokeActionProps {
  // these 3 same as now
  readonly output?: codepipeline.Artifact;
  readonly stateMachine: stepfunction.IStateMachine;
  readonly executionNamePrefix?: string;
 
  // exactly one of inputFile and inputObject is required - we check that in the  StepFunctionsInvokeAction class
  readonly inputFile?: codepipeline.ArtifactPath;
  readonly inputObject: object;
}
```

Let me know which one you like better!